manager/node.go

```diff
@@ -145,29 +103,6 @@ func (m *VolumeManager) DiskUpdate(name string, updateDisks map[string]longhorn.
 		return nil, err
 	}
 
-	originDisks := node.Spec.Disks
-
-	for name, uDisk := range updateDisks {
-		if uDisk.StorageReserved < 0 {
-			return nil, fmt.Errorf("update disk on node %v error: The storageReserved setting of disk %v(%v) is not valid, should be positive and no more than storageMaximum and storageAvailable", name, name, uDisk.Path)
-		}
-
-		tags, err := util.ValidateTags(uDisk.Tags)
-		if err != nil {
-			return nil, err
-		}
-		uDisk.Tags = tags
-		updateDisks[name] = uDisk
-	}
-
-	// delete disks
-	for name, oDisk := range originDisks {
-		if _, ok := updateDisks[name]; !ok {
-			if oDisk.AllowScheduling || node.Status.DiskStatus[name].StorageScheduled != 0 {
-				return nil, fmt.Errorf("Delete Disk on node %v error: Please disable the disk %v and remove all replicas first ", name, oDisk.Path)
-			}
-		}
-	}
 	node.Spec.Disks = updateDisks
```

Comment:
@shuo-wu 
I will create a ticket for `volumeManager` refactoring, okay?

Reply:
Sure. But you still need to move the validation/mutation logic to the webhook in this PR......

Reply:
Yep.

Reply:
@shuo-wu 
Some **update** validation and mutation cannot be moved to the webhook, because there are conflicts between the actions. https://github.com/longhorn/longhorn/issues/3562#issuecomment-1041214241
Thus, I only move the general validation and mutation logics to the webhook.
WDYT?

Reply:
OK, YES. We just need to move the general checks to the webhook.